include/revng/ADT/GenericGraph.h

```diff
@@ -996,6 +996,124 @@ class MutableEdgeNode
     return llvm::make_range(predecessors_begin(), predecessors_end());
   }
 
+private:
+  /// Custom child iterator which iterates over the concatenation of the
+  /// successors and predecessors of a node. This is used for the `Undirected`
+  /// `GraphTraits` specialization. This is done taking inspiration from the
+  /// internals of `llvm::concat` implementation, but overcoming the limitation
+  /// causing by the return `EdgeView` going of out scope.
```

Comment:
1. > overcoming the limitation causing by the return `EdgeView` going of out scope

   This does not provide me with enough information to understand why all of this is necessary.
   Can you add an example of how you'd use `llvm::concat` highlighting what the problem would be?

 2. How does `Proxy` solves the problem?
 3. Is there a way to have a `revng::concat` that does not suffers from `llvm::concat` limitation? If so, leave a TODO about this.

Reply:
I agree the explanation was not well laid out for one that did not see the problem in its entirety. I rewrite and centralized the explanation here https://github.com/revng/revng/blob/f14ffdc32258b98198148f45162302e6cbecc7c1/include/revng/Support/Undirected.h#L28